sql/analyzer/batch.go

```diff
@@ -119,9 +119,11 @@ func (b *Batch) evalOnce(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope
 		a.PushDebugContext(rule.Id.String())
 		next, same, err = rule.Apply(ctx, a, prev, scope, sel)
 		allSame = same && allSame
-		if next != nil && !same {
-			a.LogDiff(prev, next)
+		if next != nil {
 			a.LogNode(next)
+			// We should only do this if the result has changed, but some rules currently misbehave and falsely report nothing
```

Comment:
abbreviating the output is so much more useful, even though catalog, subqueries, unions, ctes falsely report a change when there wasn't

Reply:
We definitely do not want to abbreviate output if somebody sets the debug and verbose flags on the analyzer

Reply:
"Abbreviate the output" = print the plan once not 300 times